db/lib/database-access.js

```diff
@@ -0,0 +1,101 @@
+/** @import { BlockEvent } from '../../backend/lib/rpc-service/data-types.js' */
+/** @import { Static } from '@sinclair/typebox' */
+/** @import {Queryable} from '@filecoin-station/deal-observer-db' */
+import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js'
+import { Value } from '@sinclair/typebox/value'
+
+/**
+ * @param {Queryable} pgPool
+ * @returns {Promise<ActiveDealDbEntry | null>}
+ */
+export async function fetchDealWithHighestActivatedEpoch (pgPool) {
+  const query = 'SELECT * FROM active_deals ORDER BY activated_at_epoch DESC LIMIT 1'
+  const result = await loadDeals(pgPool, query)
+  return result.length > 0 ? result[0] : null
+}
+
+/**
+   * @param {Static<typeof BlockEvent>[]} activeDeals
+   * @param {Queryable} pgPool
+   * @returns {Promise<void>}
+   * */
+export async function storeActiveDeals (activeDeals, pgPool) {
```

Comment:
Let's modify all data-access functions to accept `pgPool` as the first argument please, so that we have a consistent pattern used by all functions, and it's possible to use optional arguments.

Right now, there is inconsistency - `storeActiveDeals (activeDeals, pgPool)` vs `loadDeals (pgPool, query)`.

An illustration of an optional parameter:

```js
async function loadDeals (pgPool, query = 'SELECT * FROM active_deals') {
  const result = (await pgPool.query(query)).rows.map(deal => {
    return Value.Parse(ActiveDealDbEntry, deal)
  }
  )
  return result
}

console.log(await loadDeals(pgPool))
```